src/flashbar/collapsible-flashbar.tsx

```diff
@@ -0,0 +1,392 @@
+  const itemCountElementId = useUniqueId('item-count');
+
+  useEffect(() => {
+    if (items.length <= maxNonCollapsibleItems) {
+      setIsFlashbarStackExpanded(false);
+    }
+  }, [items.length]);
+
+  const animateFlash = !isReducedMotion;
+
+  function toggleCollapseExpand() {
+    sendToggleMetric(items.length, !isFlashbarStackExpanded);
+    if (!isReducedMotion) {
+      prepareAnimations();
+    }
+    setIsFlashbarStackExpanded(prev => !prev);
+  }
+
+  const updateBottomSpacing = useCallback(() => {
+    // Allow vertical space between Flashbar and page bottom only when the Flashbar is reaching the end of the page,
+    // otherwise avoid spacing with eventual sticky elements below.
+    const listElement = listElementRef?.current;
+    const flashbar = listElement?.parentElement;
+    if (listElement && flashbar && listTop !== undefined) {
+      const bottom = listTop + listElement.clientHeight;
+      const windowHeight = window.innerHeight;
+      flashbar.style.paddingBottom = '';
+      if (isFlashbarStackExpanded && bottom + parseInt(getComputedStyle(flashbar).paddingBottom) <= windowHeight) {
+        flashbar.style.paddingBottom = '0';
+      }
+    }
+  }, [isFlashbarStackExpanded, listTop]);
+
+  useEffect(() => {
+    window.addEventListener('resize', updateBottomSpacing);
+    return () => window.removeEventListener('resize', updateBottomSpacing);
+  }, [updateBottomSpacing]);
+
+  const { i18nStrings } = restProps;
+
+  const verifyStringPresence = useCallback(
+    (parameterName: keyof StackedFlashbarProps.I18nStrings) => {
+      if (i18nStrings && !i18nStrings[parameterName]) {
+        warnOnce(
+          componentName,
+          'Using the `collapsible` option requires passing a `' +
+            parameterName +
+            '` parameter inside the `i18nStrings` object.'
+        );
+      }
+    },
+    [i18nStrings]
+  );
+
+  useEffect(() => {
+    if (!i18nStrings) {
+      warnOnce(componentName, 'Using the `collapsible` option requires passing an `i18nStrings` object.');
+      return;
+    }
+    verifyStringPresence('ariaLabel');
+    verifyStringPresence('toggleButtonText');
+    verifyStringPresence('toggleButtonAriaLabel');
+    for (const { labelName } of counterTypes) {
+      verifyStringPresence(labelName);
+    }
+  }, [i18nStrings, verifyStringPresence]);
+
+  useLayoutEffect(() => {
+    // When `useLayoutEffect` is called, the DOM is updated but has not been painted yet,
+    // so it's a good moment to trigger animations that will make calculations based on old and new DOM state.
+    // The old state is kept in `initialAnimationState`
+    // and the new state can be retrieved from the current DOM elements.
+
+    if (listTop === undefined) {
+      const listElement = listElementRef?.current;
+      if (listElement) {
+        setListTop(listElement.getBoundingClientRect().y);
+      }
+    }
+
+    if (initialAnimationState) {
+      updateBottomSpacing();
+      animate({
```

Comment:
Why do we need to use these special FLIP animations? Can we not use CSS and React Transitions?

Reply:
Because the Flash could be in different places (screen coordinates) both in the initial and the final state, so we would not know the amount of translation that we should apply.

For enter and leave animations we keep using CSS and React Transitions —this keeps being the preferred method when possible.